teuthology/util/scanner.py

```diff
@@ -0,0 +1,148 @@
+import logging
+import yaml
+from collections import defaultdict
+from lxml import etree
+
+log = logging.getLogger(__name__)
+
+
+class Scanner():
+    def __init__(self, remote=None) -> None:
+        self.summary_data = []
+        self.remote = remote
+
```

Comment:
This is nit, but I think the order of the functions should be:
1. _parse
2. scan_file
3. scan_all_files

Just for better readability